service/src/main/java/uk/nhs/adaptors/gp2gp/common/storage/StorageConnectorService.java

```diff
@@ -26,7 +26,17 @@ public void uploadWithMetadata(StorageDataWrapper response) {
         String jsonStringResponse = objectMapper.writeValueAsString(response);
         var responseBytes = jsonStringResponse.getBytes(UTF_8);
         var responseInputStream = new ByteArrayInputStream(responseBytes);
+
         storageConnector.uploadToStorage(responseInputStream, responseBytes.length,
             response.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
     }
+
+    @SneakyThrows(JsonProcessingException.class)
+    public void uploadDocument(String documentName, StorageDataWrapper response) {
+        String jsonStringResponse = objectMapper.writeValueAsString(response);
```

Comment:
We could reuse this method and remove the one above by making structured task pass the file name, you'll need to slight alter the test